### ISSUE TITLE
Add support for ISO 8601 format duration strings

### ISSUE THREAD

**Issue by stevdompl** [jkroso/parse-duration#30]
Hi,

is it possible to add support for ISO 8601 formatted duration strings? 
I have to add moment.js library as a dependency (besides yours) just for parsing this format.

More details: https://en.wikipedia.org/wiki/ISO_8601#Durations

**Comment by SimonJang** [comment]
@stevdompl you could use [ISO8601-duration](https://github.com/tolu/ISO8601-duration) for that. Would be nice this library would support the format as well.

**Comment by dy** [comment]
It would be trivial to implement by just enabling upcase units. But in ISO8601 there's ambiguity of `M` - it can both mean month and minute, depending on preceding `P` or `T`. That seems to be a bit custom parsing algorithm.
Can you give an example a use-case where parsing both ISO 8601 and human durations is worthy?
I wonder is that conceptually a good fit for this module.

**Comment by stevdompl** [comment]
Hi, you see my app is build as an middleware that helps me with migration to my new API. 

We got a few legacy sources, two new ones that use the ISO8601 format. On the older ones the duration was saved as a text value in human-readable format (that your lib parses), the other two use the C# TimeSpan format.

I've used your lib in conjunction with moment.js and got it working, but the moment.js dependency is very heavy IMHO.
